lib/Server.js

```
@@ -3412,39 +3412,109 @@ class Server {
     const isWin = path.sep === "\\";
     const toPosix = (/** @type {string} */ filePath) =>
       isWin ? filePath.split(path.sep).join("/") : filePath;
-    const toNative = (/** @type {string} */ filePath) =>
-      isWin ? filePath.split("/").join(path.sep) : filePath;
+    const cwd = watchOptions.cwd || process.cwd();
+    const absolute = (/** @type {string} */ item) =>
+      toPosix(path.resolve(cwd, item));
+    const matchesGlob = (
+      /** @type {string} */ file,
+      /** @type {string} */ pattern,
+    ) => path.matchesGlob(absolute(file), absolute(pattern));
```

ESLint

[error] The 'path.matchesGlob' is still an experimental feature and is not supported until Node.js 24.8.0 (backported: ^22.20.0). The configured version range is '>= 22.15.0'.

(n/no-unsupported-features/node-builtins)